File: src/main/java/pl/dmichalski/algorithms/_7_binary_search/SearchService.java
package pl.dmichalski.algorithms._7_binary_search;

class SearchService {

    /**
     * O(n) time complexity
     */
    Integer search(int[] sortedValues, int numberToSearch) {
        if (sortedValues == null) {
            return -1;
        }

        for (int i = 0; i < sortedValues.length; i++) {
            if (sortedValues[i] == numberToSearch) {
                return i;
            }
        }

        return -1;
    }
}

File: src/main/kotlin/pl/dmichalski/algorithms/_7_binary_search/SearchService.kt
package pl.dmichalski.algorithms._7_binary_search

internal class SearchService {

    /**
     * O(n) time complexity
     */
    fun search(sortedValues: IntArray, numberToSearch: Int): Int {
        for (i in sortedValues.indices) {
            if (sortedValues[i] == numberToSearch) {
                return i
            }
        }

        return -1
    }

}
